packages/wonder-stuff-core/src/__tests__/kind-error.test.js

```diff
@@ -240,6 +273,30 @@ describe("KindError", () => {
                 // Assert
                 expect(error.message).toBe(combinedErrorInfo.message);
             });
+
+            it("should have a combined message that cascades all causes", () => {
+                // Arrange
+                const cause1 = new Error("CAUSE_MESSAGE");
+                const cause2 = new KindError(
+                    "CAUSE_2_MESSAGE",
+                    Errors.Unknown,
+                    {cause: cause1},
+                );
+
+                // Act
+                const result = new KindError("ROOT_MESSAGE", Errors.Unknown, {
+                    cause: cause2,
+                });
+
+                // Assert
+                expect(result.message).toMatchInlineSnapshot(`
+                    "UnknownError: ROOT_MESSAGE
+                    	caused by
+                    		UnknownError: UnknownError: CAUSE_2_MESSAGE
+                    	caused by
+                    		Error: CAUSE_MESSAGE"
+                `);
```

Comment:
@lillialexis here's the test I was talking about :)

Reply:
yay!